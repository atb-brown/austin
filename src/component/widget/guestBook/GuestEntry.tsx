import { GuestEntryProps } from "./GuestEntryProps";
import React from "react";

export const guestEntryGap = 3;

/**
 * This component is a single entry for a single guest. Multiple entries will make up a guest book.
 *
 * @param {GuestEntryProps} guestEntry The properties/data of a single guest.
 * @return {GuestEntry}
 */
export default function GuestEntry(guestEntry: GuestEntryProps) {
  const bgColor = "SteelBlue";
  const bRadius = 15;
  const { name, website, message, visitDate } = guestEntry;
  return (
    <div
      style={{
        display: "flex",
        width: "95%",
        height: "85px",
      }}
    >
      <div
        style={{
          flex: 2,
          flexDirection: "column",
          display: "flex",
          backgroundColor: bgColor,
          borderTopLeftRadius: bRadius,
          borderBottomLeftRadius: bRadius,
          justifyContent: "space-evenly",
        }}
      >
        <div
          style={{
            fontSize: "75%",
            justifyContent: "center",
          }}
        >
          {name}
        </div>
        <div
          style={{
            justifyContent: "center",
            fontSize: "50%",
          }}
        >
          {website}
        </div>
      </div>
      <div
        style={{
          flex: 7,
          display: "flex",
          justifyContent: "center",
          backgroundColor: bgColor,
          marginLeft: guestEntryGap,
          marginRight: guestEntryGap,
        }}
      >
        <div
          style={{
            fontSize: "75%",
            display: "flex",
            alignItems: "center",
            height: "100%",
            width: "98%",
          }}
        >
          {message}
        </div>
      </div>
      <div
        style={{
          flex: 1,
          backgroundColor: bgColor,
          fontSize: "60%",
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          borderTopRightRadius: bRadius,
          borderBottomRightRadius: bRadius,
        }}
      >
        {visitDate}
      </div>
    </div>
  );
}
