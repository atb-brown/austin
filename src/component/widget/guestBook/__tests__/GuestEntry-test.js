import GuestEntry from "../GuestEntry";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

it("renders to match snapshot", () => {
  const tree = renderer.create(
    <GuestEntry
      name="austin"
      message="Hello World!"
      visitDate="2023-11-26"
      website="https://atb-brown.github.io/austin/"
      isFirst={true}
      isLast={true}
    />,
  );

  expect(tree).toMatchSnapshot();
});

it("top row entry", () => {
  const tree = renderer.create(
    <GuestEntry
      name="austin"
      message="Hello World!"
      visitDate="2023-11-26"
      website="https://atb-brown.github.io/austin/"
      isFirst={true}
      isLast={false}
    />,
  );

  expect(tree).toMatchSnapshot();
});

it("bottom row entry", () => {
  const tree = renderer.create(
    <GuestEntry
      name="austin"
      message="Hello World!"
      visitDate="2023-11-26"
      website="https://atb-brown.github.io/austin/"
      isFirst={false}
      isLast={true}
    />,
  );

  expect(tree).toMatchSnapshot();
});

it("renders without properties", () => {
  const tree = renderer.create(<GuestEntry />);

  expect(tree).toMatchSnapshot();
});
