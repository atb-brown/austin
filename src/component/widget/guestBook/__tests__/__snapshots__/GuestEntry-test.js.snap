// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bottom row entry 1`] = `
<div
  style={
    {
      "borderBottomLeftRadius": 15,
      "borderBottomRightRadius": 15,
      "borderTopLeftRadius": 0,
      "borderTopRightRadius": 0,
      "display": "flex",
      "height": "85px",
      "overflow": "hidden",
      "width": "95%",
    }
  }
>
  <div
    className="Guests-Visual"
    style={
      {
        "flex": 2,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
      }
    }
  >
    <div
      className="Guests-Text-container"
      style={
        {
          "fontSize": "75%",
        }
      }
    >
      austin
    </div>
    <div
      className="Guests-Text-container"
      style={
        {
          "fontSize": "50%",
          "justifyContent": "center",
        }
      }
    >
      https://atb-brown.github.io/austin/
    </div>
  </div>
  <div
    className="Guests-Visual Guests-Text-Container"
    style={
      {
        "flex": 7,
        "marginLeft": 3,
        "marginRight": 3,
      }
    }
  >
    <div
      className="Guests-Text-Container"
      style={
        {
          "fontSize": "75%",
          "height": "100%",
          "width": "98%",
        }
      }
    >
      Hello World!
    </div>
  </div>
  <div
    className="Guests-Visual Guests-Text-Container"
    style={
      {
        "flex": 1,
        "fontSize": "60%",
      }
    }
  >
    2023-11-26
  </div>
</div>
`;

exports[`renders to match snapshot 1`] = `
<div
  style={
    {
      "borderBottomLeftRadius": 15,
      "borderBottomRightRadius": 15,
      "borderTopLeftRadius": 15,
      "borderTopRightRadius": 15,
      "display": "flex",
      "height": "85px",
      "overflow": "hidden",
      "width": "95%",
    }
  }
>
  <div
    className="Guests-Visual"
    style={
      {
        "flex": 2,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
      }
    }
  >
    <div
      className="Guests-Text-container"
      style={
        {
          "fontSize": "75%",
        }
      }
    >
      austin
    </div>
    <div
      className="Guests-Text-container"
      style={
        {
          "fontSize": "50%",
          "justifyContent": "center",
        }
      }
    >
      https://atb-brown.github.io/austin/
    </div>
  </div>
  <div
    className="Guests-Visual Guests-Text-Container"
    style={
      {
        "flex": 7,
        "marginLeft": 3,
        "marginRight": 3,
      }
    }
  >
    <div
      className="Guests-Text-Container"
      style={
        {
          "fontSize": "75%",
          "height": "100%",
          "width": "98%",
        }
      }
    >
      Hello World!
    </div>
  </div>
  <div
    className="Guests-Visual Guests-Text-Container"
    style={
      {
        "flex": 1,
        "fontSize": "60%",
      }
    }
  >
    2023-11-26
  </div>
</div>
`;

exports[`renders without properties 1`] = `
<div
  style={
    {
      "borderBottomLeftRadius": 0,
      "borderBottomRightRadius": 0,
      "borderTopLeftRadius": 0,
      "borderTopRightRadius": 0,
      "display": "flex",
      "height": "85px",
      "overflow": "hidden",
      "width": "95%",
    }
  }
>
  <div
    className="Guests-Visual"
    style={
      {
        "flex": 2,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
      }
    }
  >
    <div
      className="Guests-Text-container"
      style={
        {
          "fontSize": "75%",
        }
      }
    />
    <div
      className="Guests-Text-container"
      style={
        {
          "fontSize": "50%",
          "justifyContent": "center",
        }
      }
    />
  </div>
  <div
    className="Guests-Visual Guests-Text-Container"
    style={
      {
        "flex": 7,
        "marginLeft": 3,
        "marginRight": 3,
      }
    }
  >
    <div
      className="Guests-Text-Container"
      style={
        {
          "fontSize": "75%",
          "height": "100%",
          "width": "98%",
        }
      }
    />
  </div>
  <div
    className="Guests-Visual Guests-Text-Container"
    style={
      {
        "flex": 1,
        "fontSize": "60%",
      }
    }
  />
</div>
`;

exports[`top row entry 1`] = `
<div
  style={
    {
      "borderBottomLeftRadius": 0,
      "borderBottomRightRadius": 0,
      "borderTopLeftRadius": 15,
      "borderTopRightRadius": 15,
      "display": "flex",
      "height": "85px",
      "overflow": "hidden",
      "width": "95%",
    }
  }
>
  <div
    className="Guests-Visual"
    style={
      {
        "flex": 2,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
      }
    }
  >
    <div
      className="Guests-Text-container"
      style={
        {
          "fontSize": "75%",
        }
      }
    >
      austin
    </div>
    <div
      className="Guests-Text-container"
      style={
        {
          "fontSize": "50%",
          "justifyContent": "center",
        }
      }
    >
      https://atb-brown.github.io/austin/
    </div>
  </div>
  <div
    className="Guests-Visual Guests-Text-Container"
    style={
      {
        "flex": 7,
        "marginLeft": 3,
        "marginRight": 3,
      }
    }
  >
    <div
      className="Guests-Text-Container"
      style={
        {
          "fontSize": "75%",
          "height": "100%",
          "width": "98%",
        }
      }
    >
      Hello World!
    </div>
  </div>
  <div
    className="Guests-Visual Guests-Text-Container"
    style={
      {
        "flex": 1,
        "fontSize": "60%",
      }
    }
  >
    2023-11-26
  </div>
</div>
`;
